src/properties.rs

```diff
@@ -1,7 +1,24 @@
-use crate::{de::PacketParser, ser::ReversedPacketWriter, ProtocolError as Error};
+use crate::{
+    de::PacketParser,
+    ser::ReversedPacketWriter,
+    ProtocolError as Error
+};
 
+use serde::Deserialize;
 use enum_iterator::IntoEnumIterator;
 
+impl From<u32> for PropertyIdentifier {
```

Comment:
I think in general [`num_enum`](https://docs.rs/num_enum/latest/num_enum/) is better for these things. And it is pretty customizable.